packages/@okta/vuepress-site/docs/guides/implement-oauth-for-okta-serviceapp/create-publicprivate-keypair/index.md

```diff
@@ -0,0 +1,37 @@
+---
+title: Create a public/private key pair
+---
+
+The `private_key_jwt` client authentication method is the only supported method for OAuth service apps that want to get access tokens with Okta scopes.
+
+The private key that you use to sign the JWT must have the corresponding public key registered in the [JWKSet](/docs/reference/api/oauth-clients/#json-web-key-set) of the OAuth service app. We recommend generating the public/private key pair first before creating the OAuth service app.
+
+1. Use a tool such as this [JSON Web Key Generator](https://mkjwk.org/) to generate a JWKS public and private key pair for testing. Okta supports both RSA and Elliptic Curve (EC) keys. In this example, we are selecting **RSA** as the encryption algorithm. Select the following values:
+
+    * Key size &mdash; 2048
+    * Key use &mdash; signature
+    * Algorithm &mdash; RSA256
+    * Key ID &mdash; (Optional) This can be any random value.
+
+> **Note:** Use the JSON Web Key Generator link to generate JWKS key pairs for testing purposes only. For a production use case, use your own [internal instance](https://github.com/mitreid-connect/mkjwk.org) of the key pair generator.
+
+2. The JSON Web Key Generator tool extracts the public key from the private key automatically. For testing purposes, copy the Public Key that is provided.
```

Comment:
I don't think the public key is extracted from the private key.
It should be "extracts the public key from the key pair automatically"
@arvindkrishnakumar-okta, @bretterer, @swiftone  - any thoughts?

Reply:
I think you are correct. :)